cli/cmd/edges.go

```diff
@@ -260,58 +278,75 @@ func writeEdgesToBuffer(rows []*pb.Edge, w *tabwriter.Writer, options *edgesOpti
 		}
 	}
 
-	// sorting edgeRows by key for alphabetical listing
+	// ordering the rows first by SRC/DST namespace, then by SRC/DST resource
 	sort.Slice(edgeRows, func(i, j int) bool {
-		keyI := edgeRows[i].src + edgeRows[i].dst
-		keyJ := edgeRows[j].src + edgeRows[j].dst
+		keyI := edgeRows[i].srcNamespace + edgeRows[i].dstNamespace + edgeRows[i].src + edgeRows[i].dst
+		keyJ := edgeRows[j].srcNamespace + edgeRows[j].dstNamespace + edgeRows[j].src + edgeRows[j].dst
 		return keyI < keyJ
 	})
 
 	switch options.outputFormat {
-	case tableOutput:
+	case tableOutput, wideOutput:
 		if len(edgeRows) == 0 {
 			fmt.Fprintln(os.Stderr, "No edges found.")
 			os.Exit(0)
 		}
-		printEdgeTable(edgeRows, w, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength)
+		printEdgeTable(edgeRows, w, maxSrcLength, maxSrcNamespaceLength, maxDstLength, maxDstNamespaceLength, maxClientLength, maxServerLength, maxMsgLength, options.outputFormat)
 	case jsonOutput:
 		printEdgesJSON(edgeRows, w)
 	}
 }
 
-func printEdgeTable(edgeRows []edgeRow, w *tabwriter.Writer, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength int) {
+func printEdgeTable(edgeRows []edgeRow, w *tabwriter.Writer, maxSrcLength, maxSrcNamespaceLength, maxDstLength, maxDstNamespaceLength, maxClientLength, maxServerLength, maxMsgLength int, outputFormat string) {
 	srcTemplate := fmt.Sprintf("%%-%ds", maxSrcLength)
 	dstTemplate := fmt.Sprintf("%%-%ds", maxDstLength)
+	srcNamespaceTemplate := fmt.Sprintf("%%-%ds", maxSrcNamespaceLength)
+	dstNamespaceTemplate := fmt.Sprintf("%%-%ds", maxDstNamespaceLength)
+	msgTemplate := fmt.Sprintf("%%-%ds", maxMsgLength)
 	clientTemplate := fmt.Sprintf("%%-%ds", maxClientLength)
 	serverTemplate := fmt.Sprintf("%%-%ds", maxServerLength)
-	msgTemplate := fmt.Sprintf("%%-%ds", maxMsgLength)
 
 	headers := []string{
 		fmt.Sprintf(srcTemplate, srcHeader),
 		fmt.Sprintf(dstTemplate, dstHeader),
-		fmt.Sprintf(clientTemplate, clientHeader),
-		fmt.Sprintf(serverTemplate, serverHeader),
+		fmt.Sprintf(srcNamespaceTemplate, srcNamespaceHeader),
+		fmt.Sprintf(dstNamespaceTemplate, dstNamespaceHeader),
 		fmt.Sprintf(msgTemplate, msgHeader),
 	}
 
+	if outputFormat == wideOutput {
+		// insert CLIENT_ID and SERVER_ID headers before final SECURED header
+		headers = append(headers[:len(headers)-1], append([]string{
+			fmt.Sprintf(clientTemplate, clientHeader),
+			fmt.Sprintf(serverTemplate, serverHeader),
+		}, headers[len(headers)-1:]...)...)
+	}
+
```

Comment:
May I suggest a simpler approach here?
```go
headers := []string{
		fmt.Sprintf(srcTemplate, srcHeader),
		fmt.Sprintf(dstTemplate, dstHeader),
		fmt.Sprintf(srcNamespaceTemplate, srcNamespaceHeader),
		fmt.Sprintf(dstNamespaceTemplate, dstNamespaceHeader),
	}
	if outputFormat == wideOutput {
		headers = append(headers, fmt.Sprintf(clientTemplate, clientHeader), fmt.Sprintf(serverTemplate, serverHeader))
	}
	headers = append(headers, fmt.Sprintf(msgTemplate, msgHeader)+"\t")
```

Reply:
👍 Much simpler, thanks for the suggestion!